# Notebook for running all the regressions using OLS for every group of cause--sex, as this is for age standadized data

## Based on the previous analysis, we use the age groups upto 74 years and logit SDI as the predictor rather than SDI, along with the HAQ indices obtained for GBD round 5

## Loading causes based on highest expenditure data (based on Joe's list and COD methods)

In [1]:
import pandas as pd
cause_list_final = pd.read_excel('/ihme/homes/arjuns13/notebooks/Documents/Data/CandidateCauses_Criteria_withCODEM_andDALYs_latest2ndJune.xlsx')

In [2]:
cause_list_final.head()

Unnamed: 0 DEX ranking by total expenditure  GBD cause_id  \
0          61                               75         298.0   
1           0                                1         587.0   
2           1                                2         493.0   
3           3                                5         697.0   
4           9                               12         509.0   

                   DEX descriptive label  \
0                               HIV/AIDS   
1                      Diabetes mellitus   
2                 Ischemic heart disease   
3                                  Falls   
4  Chronic obstructive pulmonary disease   

  Preliminary amenable burden analysis conducted  Comment  \
0                                             NaN     NaN   
1                                             Yes     NaN   
2                                             NaN     NaN   
3                                             Yes     NaN   
4                                             Yes     NaN   

   Number of GBD round 5 global CODEM models in which HAQ index was included  \
0                                                NaN                           
1                                              694.0                           
2                                              372.5                           
3                                              352.5                           
4                                              532.5                           

   Number of GBD round 5data rich CODEM models in which HAQ index was included  \
0                                                NaN                             
1                                              553.0                             
2                                              608.5                             
3                                               30.0                             
4                                              353.0                             

  Was HAQ index included in nonfatal model?  Ranking DALYs in US Priority  \
0                                       NaN                   53        *   
1           Yes, proportion with foot ulcer                    6        4   
2                                       Yes                    1        4   
3                           Yes, % treated                    18        4   
4                                      Yes                     4        4   

   Total DALYs in the United States, GBD round 5, year = 2016  \
0                                       4.099518e+05            
1                                       3.814559e+06            
2                                       7.854095e+06            
3                                       1.899736e+06            
4                                       4.613470e+06            

   DALYs per 100,000 people in the United States, GBD round 5, year = 2016  \
0                                         126.970345                         
1                                        1181.445965                         
2                                        2432.571827                         
3                                         588.386632                         
4                                        1428.884710                         

   Total DALYs globally, GBD round 5, year = 2016  \
0                                    5.904798e+07   
1                                    6.651521e+07   
2                                    1.688977e+08   
3                                    3.551392e+07   
4                                    8.059322e+07   

   DALYs per 100000  people globally, GBD round 5, year = 2016  \
0                                         781.627137             
1                                         880.471923             
2                                        2235.725150             
3                                         470.103220             
4                                       

In [3]:
cause_list_final.shape

(149, 19)

In [4]:
cause_ids_final = cause_list_final['GBD cause_id']

In [5]:
cause_ids_final = cause_ids_final.dropna()

In [6]:
cause_ids_final = cause_ids_final.astype('int')

### getting the values computed earlier - for HAQ frontiers, Logit_SDIs and older DALYs (without cause IDs)

In [7]:
haqFront_sdi_daly_data = pd.read_csv('/ihme/homes/arjuns13/notebooks/Documents/Data/haq_sdi_frontier_estimates_ageStd_aggregatedCauses_allMostDetLocs.csv')

In [8]:
haqFront_sdi_daly_data.head()

Unnamed: 0  V1  location_id  logit_sdi  age_group_id    ln_haq      fit2
0           1   0            7   0.110348            27  3.792486  3.757823
1           2   1            8   1.820419            27  4.367523  4.668259
2           3   2           10  -0.093432            27  3.509703  3.649331
3           4   3           12   0.046388            27  3.245486  3.723771
4           5   4           13   1.137170            27  4.000794  4.304499

In [9]:
final = haqFront_sdi_daly_data

In [10]:
final['haq'] = 10**(final['ln_haq'])

In [11]:
final['exp_fit'] = 10**(final['fit2'])

In [12]:
final.loc[final['exp_fit'] > 100.0, 'exp_fit']=100


In [13]:
final['exp_fit'].describe()

count    823.0
mean     100.0
std        0.0
min      100.0
25%      100.0
50%      100.0
75%      100.0
max      100.0
Name: exp_fit, dtype: float64

In [14]:
final['haq_over_frontier'] = final['haq']/final['exp_fit']

## End of the copied part from the previously done analysis

In [15]:
final.head()

Unnamed: 0  V1  location_id  logit_sdi  age_group_id    ln_haq      fit2  \
0           1   0            7   0.110348            27  3.792486  3.757823   
1           2   1            8   1.820419            27  4.367523  4.668259   
2           3   2           10  -0.093432            27  3.509703  3.649331   
3           4   3           12   0.046388            27  3.245486  3.723771   
4           5   4           13   1.137170            27  4.000794  4.304499   

            haq  exp_fit  haq_over_frontier  
0   6201.348166    100.0          62.013482  
1  23308.955739    100.0         233.089557  
2   3233.720992    100.0          32.337210  
3   1759.891080    100.0          17.598911  
4  10018.298652    100.0         100.182987

## To be merged (on Location Id and Age Group ID) with the DALY data coming from the central funcs - here I end up keeping LogitSDI, rather than SDI, which is what I used for the 2nd June run

In [16]:
haq_sdi = final[['location_id', 'age_group_id', 'logit_sdi','exp_fit','haq','haq_over_frontier']]

In [17]:
haq_sdi.head()

location_id  age_group_id  logit_sdi  exp_fit           haq  \
0            7            27   0.110348    100.0   6201.348166   
1            8            27   1.820419    100.0  23308.955739   
2           10            27  -0.093432    100.0   3233.720992   
3           12            27   0.046388    100.0   1759.891080   
4           13            27   1.137170    100.0  10018.298652   

   haq_over_frontier  
0          62.013482  
1         233.089557  
2          32.337210  
3          17.598911  
4         100.182987

In [18]:
haq_sdi.shape

(823, 6)

## Getting all the 823 locations and the 146 causes from above data to enter as input into the get_outputs central function

In [19]:
location_list = list(haq_sdi.location_id.unique())

In [20]:
len(location_list)

823

In [21]:
cause_list = list(cause_ids_final.unique())

In [22]:
len(cause_list)

146

In [23]:
from db_queries import get_outputs as go 
dalys_per_cause_rate = go("cause", cause_id=cause_list, metric_id=[3],\
                     measure_id=2, gbd_round_id=5,year_id=[2016],sex_id=[1,2],location_id=location_list,\
                    age_group_id = "all")
dalys_per_cause_rate.head()

age_group_id  cause_id  location_id  measure_id  metric_id  sex_id  \
0             2       297            7           2          3       1   
1             2       298            7           2          3       1   
2             2       302            7           2          3       1   
3             2       321            7           2          3       1   
4             2       322            7           2          3       1   

   year_id        acause  age_group_name  \
0     2016            tb  Early Neonatal   
1     2016           hiv  Early Neonatal   
2     2016      diarrhea  Early Neonatal   
3     2016  intest_other  Early Neonatal   
4     2016           lri  Early Neonatal   

                             cause_name expected location_name  \
0                          Tuberculosis    False   North Korea   
1                              HIV/AIDS    False   North Korea   
2                    Diarrheal diseases    False   North Korea   
3  Other intestinal infectious diseases    False   North Korea   
4          Lower respiratory infections    False   North Korea   

                             measure_name metric_name   sex           val  \
0  DALYs (Disability-Adjusted Life Years)        Rate  Male  5.964052e-07   
1  DALYs (Disability-Adjusted Life Years)        Rate  Male  8.583118e-06   
2  DALYs (Disability-Adjusted Life Years)        Rate  Male  5.872325e-02   
3  DALYs (Disability-Adjusted Life Years)        Rate  Male  0.000000e+00   
4  DALYs (Disability-Adjusted Life Years)        Rate  Male  3.201836e+00   

      upper         lower  
0  0.000001  2.937580e-07  
1  0.000059  2.354980e-07  
2  0.108532  2.859224e-02  
3  0.000000  0.000000e+00  
4  5.515528  1.658975e+00

## Just some data validation

In [24]:
dalys_per_cause_rate[(dalys_per_cause_rate.location_id == 13) & (dalys_per_cause_rate.age_group_id == 17)\
                    & (dalys_per_cause_rate.cause_id == 493)]

age_group_id  cause_id  location_id  measure_id  metric_id  sex_id  \
31311            17       493           13           2          3       1   
31457            17       493           13           2          3       2   

       year_id   acause age_group_name              cause_name expected  \
31311     2016  cvd_ihd       60 to 64  Ischemic heart disease    False   
31457     2016  cvd_ihd       60 to 64  Ischemic heart disease    False   

      location_name                            measure_name metric_name  \
31311      Malaysia  DALYs (Disability-Adjusted Life Years)        Rate   
31457      Malaysia  DALYs (Disability-Adjusted Life Years)        Rate   

          sex       val     upper     lower  
31311    Male  0.142325  0.162359  0.124818  
31457  Female  0.058360  0.069562  0.049208

### Dropping NaNs, renaming some variables

In [25]:
dalys_per_cause_rate = dalys_per_cause_rate.dropna()

In [26]:
dalys_per_cause_rate.shape

(5278722, 18)

In [27]:
dalys_per_cause_rate = dalys_per_cause_rate[['age_group_id','cause_id','location_id','sex_id','sex','age_group_name',\
                                   'cause_name','location_name','val']]

In [28]:
dalys_per_cause_rate = dalys_per_cause_rate.rename(index=str, columns={"val": "DALYs_per_capita"})

In [29]:
dalys_per_cause_rate.head()

age_group_id  cause_id  location_id  sex_id   sex  age_group_name  \
0             2       297            7       1  Male  Early Neonatal   
1             2       298            7       1  Male  Early Neonatal   
2             2       302            7       1  Male  Early Neonatal   
3             2       321            7       1  Male  Early Neonatal   
4             2       322            7       1  Male  Early Neonatal   

                             cause_name location_name  DALYs_per_capita  
0                          Tuberculosis   North Korea      5.964052e-07  
1                              HIV/AIDS   North Korea      8.583118e-06  
2                    Diarrheal diseases   North Korea      5.872325e-02  
3  Other intestinal infectious diseases   North Korea      0.000000e+00  
4          Lower respiratory infections   North Korea      3.201836e+00

In [30]:
dalys_per_cause_rate.shape

(5278722, 9)

In [31]:
dalys_per_cause = dalys_per_cause_rate

dalys_per_cause.head()

age_group_id  cause_id  location_id  sex_id   sex  age_group_name  \
0             2       297            7       1  Male  Early Neonatal   
1             2       298            7       1  Male  Early Neonatal   
2             2       302            7       1  Male  Early Neonatal   
3             2       321            7       1  Male  Early Neonatal   
4             2       322            7       1  Male  Early Neonatal   

                             cause_name location_name  DALYs_per_capita  
0                          Tuberculosis   North Korea      5.964052e-07  
1                              HIV/AIDS   North Korea      8.583118e-06  
2                    Diarrheal diseases   North Korea      5.872325e-02  
3  Other intestinal infectious diseases   North Korea      0.000000e+00  
4          Lower respiratory infections   North Korea      3.201836e+00

In [32]:
dalys_per_cause.shape

(5278722, 9)

## Don't need entries where the DALYs are zero

In [33]:
dalys_per_cause = dalys_per_cause[dalys_per_cause['DALYs_per_capita'] > 0]

In [34]:
haq_sdi_withoutAgeGroups = haq_sdi.drop(columns=['age_group_id'])

In [35]:
haq_sdi_withoutAgeGroups.head()

location_id  logit_sdi  exp_fit           haq  haq_over_frontier
0            7   0.110348    100.0   6201.348166          62.013482
1            8   1.820419    100.0  23308.955739         233.089557
2           10  -0.093432    100.0   3233.720992          32.337210
3           12   0.046388    100.0   1759.891080          17.598911
4           13   1.137170    100.0  10018.298652         100.182987

In [36]:
haq_sdi_withoutAgeGroups.shape

(823, 5)

## Merging the DALYs data with the HAQ-SDI data

In [37]:
#dalys_haq_sdi_withCauses = pd.merge(dalys_per_cause, haq_sdi, on=['location_id', 'age_group_id'])

In [38]:
dalys_haq_sdi_withCauses = pd.merge(dalys_per_cause, haq_sdi_withoutAgeGroups, on=['location_id'], how='left')

In [39]:
dalys_haq_sdi_withCauses.shape

(4426766, 13)

In [40]:
dalys_haq_sdi_withCauses.head()

age_group_id  cause_id  location_id  sex_id   sex  age_group_name  \
0             2       297            7       1  Male  Early Neonatal   
1             2       298            7       1  Male  Early Neonatal   
2             2       302            7       1  Male  Early Neonatal   
3             2       322            7       1  Male  Early Neonatal   
4             2       328            7       1  Male  Early Neonatal   

                     cause_name location_name  DALYs_per_capita  logit_sdi  \
0                  Tuberculosis   North Korea      5.964052e-07   0.110348   
1                      HIV/AIDS   North Korea      8.583118e-06   0.110348   
2            Diarrheal diseases   North Korea      5.872325e-02   0.110348   
3  Lower respiratory infections   North Korea      3.201836e+00   0.110348   
4  Upper respiratory infections   North Korea      7.761692e-03   0.110348   

   exp_fit          haq  haq_over_frontier  
0    100.0  6201.348166          62.013482  
1    100.0  6201.348166          62.013482  
2    100.0  6201.348166          62.013482  
3    100.0  6201.348166          62.013482  
4    100.0  6201.348166          62.013482

In [41]:
dalys_haq_sdi_withCauses_distinct = dalys_haq_sdi_withCauses[['age_group_id','age_group_name','cause_id','cause_name', \
                                                             'sex_id', 'sex']].drop_duplicates()

### Removing the nominal variables before running the regressions

In [42]:
dalys_haq_sdi_withCauses_forAnalysis = dalys_haq_sdi_withCauses[['age_group_id','cause_id','location_id','sex_id',\
                                                                 'DALYs_per_capita','logit_sdi','exp_fit','haq',\
                                                                 'haq_over_frontier']]

In [43]:
dalys_haq_sdi_withCauses_forAnalysis.head()

age_group_id  cause_id  location_id  sex_id  DALYs_per_capita  logit_sdi  \
0             2       297            7       1      5.964052e-07   0.110348   
1             2       298            7       1      8.583118e-06   0.110348   
2             2       302            7       1      5.872325e-02   0.110348   
3             2       322            7       1      3.201836e+00   0.110348   
4             2       328            7       1      7.761692e-03   0.110348   

   exp_fit          haq  haq_over_frontier  
0    100.0  6201.348166          62.013482  
1    100.0  6201.348166          62.013482  
2    100.0  6201.348166          62.013482  
3    100.0  6201.348166          62.013482  
4    100.0  6201.348166          62.013482

## Capping exp_fit2 to 100 for the corresponding values of exp_fit

In [44]:
dalys_haq_sdi_withCauses_forAnalysis.loc[dalys_haq_sdi_withCauses_forAnalysis['exp_fit'] > 100.0, 'exp_fit']=100

/ihme/homes/arjuns13/miniconda3/lib/python3.7/site-packages/pandas/core/indexing.py:543: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [45]:
dalys_haq_sdi_withCauses_forAnalysis.haq_over_frontier.unique()

array([ 62.01348166, 233.08955739,  32.33720992,  17.5989108 ,
       100.18298652, 141.63989944,  28.37777262,  34.2991682 ,
       122.22135935, 125.30192674,  25.97964882,  93.39555114,
        33.50507556,   7.52333683,  17.26253802,  29.73189481,
        11.29523679,  49.60730689,  22.86186629,  45.46398258,
        12.37207793, 118.83417612,  80.33557693,  90.70631385,
       107.37166588,  78.21058403,  45.94913863,  45.09423632,
        69.24833759,  73.88147818, 170.92152226, 149.43937228,
       141.39425977, 242.9364494 , 233.27459909, 187.65783498,
       158.10479894, 196.59846731, 189.22837762, 161.39352898,
       169.50499184, 196.78447996, 277.61640292, 180.44600488,
       219.0054231 , 167.48495295, 154.47135333, 110.56246956,
       118.11384125, 280.82429404, 276.96856608, 309.39870149,
       309.68483729, 294.62928293, 289.0114149 , 288.01800671,
       279.14711991, 308.55150685, 298.20194251, 278.1555956 ,
       277.26656011, 332.56422548, 317.31768895, 248.53

### Creating the groups (Cause-Id, Age-Group, Sex-Id)

In [46]:
dalys_haq_sdi_grouped =  \
dict(tuple(dalys_haq_sdi_withCauses_forAnalysis.groupby(['cause_id','age_group_id', 'sex_id'])))

### This is just how the data looks for the cause 530, age group 2, and sex 1, just as an example

In [47]:
dalys_haq_sdi_grouped[530,2,1].head()

age_group_id  cause_id  location_id  sex_id  DALYs_per_capita  \
35                2       530            7       1          0.131228   
5394              2       530            8       1          0.002186   
10755             2       530           10       1          0.141341   
16166             2       530           12       1          0.087819   
21583             2       530           13       1          0.007450   

       logit_sdi  exp_fit           haq  haq_over_frontier  
35      0.110348    100.0   6201.348166          62.013482  
5394    1.820419    100.0  23308.955739         233.089557  
10755  -0.093432    100.0   3233.720992          32.337210  
16166   0.046388    100.0   1759.891080          17.598911  
21583   1.137170    100.0  10018.298652         100.182987

In [48]:
len(dalys_haq_sdi_grouped)

5427

## Next we create a DataFrame where all the final regression results would be stored

In [49]:
dalys_haq_sdi_answers = pd.DataFrame(columns=['cause_id','age_group_id', 'sex_id',\
                                              'intercept','intercept_pValue','haq_over_frontier_coef',\
                                              'haq_over_frontier_coef_pValue','logit_sdi_coef',\
                                              'logit_sdi_coef_pValue','r2',])

In [50]:
import statsmodels.formula.api as smf
i=0
for g in dalys_haq_sdi_grouped:
    df=dalys_haq_sdi_grouped[g]
    linear_reg = smf.ols("DALYs_per_capita ~ haq_over_frontier + logit_sdi", \
                      data = df)
    linear_reg_fit = linear_reg.fit()

    
    dalys_haq_sdi_answers.loc[i] = [g[0],g[1],g[2],linear_reg_fit.params[0],linear_reg_fit.pvalues.Intercept,\
                                    linear_reg_fit.params[1],linear_reg_fit.pvalues.haq_over_frontier,\
                                    linear_reg_fit.params[2],linear_reg_fit.pvalues.logit_sdi,\
                                    linear_reg_fit.rsquared]
    i+=1
    print("Current iteration:" + str(i))

Current iteration:1
Current iteration:2
Current iteration:3
Current iteration:4
Current iteration:5
Current iteration:6
Current iteration:7
Current iteration:8
Current iteration:9
Current iteration:10
Current iteration:11
Current iteration:12
Current iteration:13
Current iteration:14
Current iteration:15
Current iteration:16
Current iteration:17
Current iteration:18
Current iteration:19
Current iteration:20
Current iteration:21
Current iteration:22
Current iteration:23
Current iteration:24
Current iteration:25
Current iteration:26
Current iteration:27
Current iteration:28
Current iteration:29
Current iteration:30
Current iteration:31
Current iteration:32
Current iteration:33
Current iteration:34
Current iteration:35
Current iteration:36
Current iteration:37
Current iteration:38
Current iteration:39
Current iteration:40
Current iteration:41
Current iteration:42
Current iteration:43
Current iteration:44
Current iteration:45
Current iteration:46
Current iteration:47
Current iteration:48
C

Current iteration:386
Current iteration:387
Current iteration:388
Current iteration:389
Current iteration:390
Current iteration:391
Current iteration:392
Current iteration:393
Current iteration:394
Current iteration:395
Current iteration:396
Current iteration:397
Current iteration:398
Current iteration:399
Current iteration:400
Current iteration:401
Current iteration:402
Current iteration:403
Current iteration:404
Current iteration:405
Current iteration:406
Current iteration:407
Current iteration:408
Current iteration:409
Current iteration:410
Current iteration:411
Current iteration:412
Current iteration:413
Current iteration:414
Current iteration:415
Current iteration:416
Current iteration:417
Current iteration:418
Current iteration:419
Current iteration:420
Current iteration:421
Current iteration:422
Current iteration:423
Current iteration:424
Current iteration:425
Current iteration:426
Current iteration:427
Current iteration:428
Current iteration:429
Current iteration:430
Current it

Current iteration:760
Current iteration:761
Current iteration:762
Current iteration:763
Current iteration:764
Current iteration:765
Current iteration:766
Current iteration:767
Current iteration:768
Current iteration:769
Current iteration:770
Current iteration:771
Current iteration:772
Current iteration:773
Current iteration:774
Current iteration:775
Current iteration:776
Current iteration:777
Current iteration:778
Current iteration:779
Current iteration:780
Current iteration:781
Current iteration:782
Current iteration:783
Current iteration:784
Current iteration:785
Current iteration:786
Current iteration:787
Current iteration:788
Current iteration:789
Current iteration:790
Current iteration:791
Current iteration:792
Current iteration:793
Current iteration:794
Current iteration:795
Current iteration:796
Current iteration:797
Current iteration:798
Current iteration:799
Current iteration:800
Current iteration:801
Current iteration:802
Current iteration:803
Current iteration:804
Current it

Current iteration:1138
Current iteration:1139
Current iteration:1140
Current iteration:1141
Current iteration:1142
Current iteration:1143
Current iteration:1144
Current iteration:1145
Current iteration:1146
Current iteration:1147
Current iteration:1148
Current iteration:1149
Current iteration:1150
Current iteration:1151
Current iteration:1152
Current iteration:1153
Current iteration:1154
Current iteration:1155
Current iteration:1156
Current iteration:1157
Current iteration:1158
Current iteration:1159
Current iteration:1160
Current iteration:1161
Current iteration:1162
Current iteration:1163
Current iteration:1164
Current iteration:1165
Current iteration:1166
Current iteration:1167
Current iteration:1168
Current iteration:1169
Current iteration:1170
Current iteration:1171
Current iteration:1172
Current iteration:1173
Current iteration:1174
Current iteration:1175
Current iteration:1176
Current iteration:1177
Current iteration:1178
Current iteration:1179
Current iteration:1180
Current ite

Current iteration:1504
Current iteration:1505
Current iteration:1506
Current iteration:1507
Current iteration:1508
Current iteration:1509
Current iteration:1510
Current iteration:1511
Current iteration:1512
Current iteration:1513
Current iteration:1514
Current iteration:1515
Current iteration:1516
Current iteration:1517
Current iteration:1518
Current iteration:1519
Current iteration:1520
Current iteration:1521
Current iteration:1522
Current iteration:1523
Current iteration:1524
Current iteration:1525
Current iteration:1526
Current iteration:1527
Current iteration:1528
Current iteration:1529
Current iteration:1530
Current iteration:1531
Current iteration:1532
Current iteration:1533
Current iteration:1534
Current iteration:1535
Current iteration:1536
Current iteration:1537
Current iteration:1538
Current iteration:1539
Current iteration:1540
Current iteration:1541
Current iteration:1542
Current iteration:1543
Current iteration:1544
Current iteration:1545
Current iteration:1546
Current ite

Current iteration:1866
Current iteration:1867
Current iteration:1868
Current iteration:1869
Current iteration:1870
Current iteration:1871
Current iteration:1872
Current iteration:1873
Current iteration:1874
Current iteration:1875
Current iteration:1876
Current iteration:1877
Current iteration:1878
Current iteration:1879
Current iteration:1880
Current iteration:1881
Current iteration:1882
Current iteration:1883
Current iteration:1884
Current iteration:1885
Current iteration:1886
Current iteration:1887
Current iteration:1888
Current iteration:1889
Current iteration:1890
Current iteration:1891
Current iteration:1892
Current iteration:1893
Current iteration:1894
Current iteration:1895
Current iteration:1896
Current iteration:1897
Current iteration:1898
Current iteration:1899
Current iteration:1900
Current iteration:1901
Current iteration:1902
Current iteration:1903
Current iteration:1904
Current iteration:1905
Current iteration:1906
Current iteration:1907
Current iteration:1908
Current ite

Current iteration:2226
Current iteration:2227
Current iteration:2228
Current iteration:2229
Current iteration:2230
Current iteration:2231
Current iteration:2232
Current iteration:2233
Current iteration:2234
Current iteration:2235
Current iteration:2236
Current iteration:2237
Current iteration:2238
Current iteration:2239
Current iteration:2240
Current iteration:2241
Current iteration:2242
Current iteration:2243
Current iteration:2244
Current iteration:2245
Current iteration:2246
Current iteration:2247
Current iteration:2248
Current iteration:2249
Current iteration:2250
Current iteration:2251
Current iteration:2252
Current iteration:2253
Current iteration:2254
Current iteration:2255
Current iteration:2256
Current iteration:2257
Current iteration:2258
Current iteration:2259
Current iteration:2260
Current iteration:2261
Current iteration:2262
Current iteration:2263
Current iteration:2264
Current iteration:2265
Current iteration:2266
Current iteration:2267
Current iteration:2268
Current ite

Current iteration:2592
Current iteration:2593
Current iteration:2594
Current iteration:2595
Current iteration:2596
Current iteration:2597
Current iteration:2598
Current iteration:2599
Current iteration:2600
Current iteration:2601
Current iteration:2602
Current iteration:2603
Current iteration:2604
Current iteration:2605
Current iteration:2606
Current iteration:2607
Current iteration:2608
Current iteration:2609
Current iteration:2610
Current iteration:2611
Current iteration:2612
Current iteration:2613
Current iteration:2614
Current iteration:2615
Current iteration:2616
Current iteration:2617
Current iteration:2618
Current iteration:2619
Current iteration:2620
Current iteration:2621
Current iteration:2622
Current iteration:2623
Current iteration:2624
Current iteration:2625
Current iteration:2626
Current iteration:2627
Current iteration:2628
Current iteration:2629
Current iteration:2630
Current iteration:2631
Current iteration:2632
Current iteration:2633
Current iteration:2634
Current ite

Current iteration:2955
Current iteration:2956
Current iteration:2957
Current iteration:2958
Current iteration:2959
Current iteration:2960
Current iteration:2961
Current iteration:2962
Current iteration:2963
Current iteration:2964
Current iteration:2965
Current iteration:2966
Current iteration:2967
Current iteration:2968
Current iteration:2969
Current iteration:2970
Current iteration:2971
Current iteration:2972
Current iteration:2973
Current iteration:2974
Current iteration:2975
Current iteration:2976
Current iteration:2977
Current iteration:2978
Current iteration:2979
Current iteration:2980
Current iteration:2981
Current iteration:2982
Current iteration:2983
Current iteration:2984
Current iteration:2985
Current iteration:2986
Current iteration:2987
Current iteration:2988
Current iteration:2989
Current iteration:2990
Current iteration:2991
Current iteration:2992
Current iteration:2993
Current iteration:2994
Current iteration:2995
Current iteration:2996
Current iteration:2997
Current ite

Current iteration:3317
Current iteration:3318
Current iteration:3319
Current iteration:3320
Current iteration:3321
Current iteration:3322
Current iteration:3323
Current iteration:3324
Current iteration:3325
Current iteration:3326
Current iteration:3327
Current iteration:3328
Current iteration:3329
Current iteration:3330
Current iteration:3331
Current iteration:3332
Current iteration:3333
Current iteration:3334
Current iteration:3335
Current iteration:3336
Current iteration:3337
Current iteration:3338
Current iteration:3339
Current iteration:3340
Current iteration:3341
Current iteration:3342
Current iteration:3343
Current iteration:3344
Current iteration:3345
Current iteration:3346
Current iteration:3347
Current iteration:3348
Current iteration:3349
Current iteration:3350
Current iteration:3351
Current iteration:3352
Current iteration:3353
Current iteration:3354
Current iteration:3355
Current iteration:3356
Current iteration:3357
Current iteration:3358
Current iteration:3359
Current ite

/ihme/homes/arjuns13/miniconda3/lib/python3.7/site-packages/statsmodels/regression/linear_model.py:1510: RuntimeWarning: divide by zero encountered in double_scalars
  return np.dot(wresid, wresid) / self.df_resid
/ihme/homes/arjuns13/miniconda3/lib/python3.7/site-packages/statsmodels/regression/linear_model.py:1543: RuntimeWarning: divide by zero encountered in double_scalars
  return 1 - self.ssr/self.centered_tss


Current iteration:3662
Current iteration:3663
Current iteration:3664
Current iteration:3665
Current iteration:3666
Current iteration:3667
Current iteration:3668
Current iteration:3669
Current iteration:3670
Current iteration:3671
Current iteration:3672
Current iteration:3673
Current iteration:3674
Current iteration:3675
Current iteration:3676
Current iteration:3677
Current iteration:3678
Current iteration:3679
Current iteration:3680
Current iteration:3681
Current iteration:3682
Current iteration:3683
Current iteration:3684
Current iteration:3685
Current iteration:3686
Current iteration:3687
Current iteration:3688
Current iteration:3689
Current iteration:3690
Current iteration:3691
Current iteration:3692
Current iteration:3693
Current iteration:3694
Current iteration:3695
Current iteration:3696
Current iteration:3697
Current iteration:3698
Current iteration:3699
Current iteration:3700
Current iteration:3701
Current iteration:3702
Current iteration:3703
Current iteration:3704
Current ite

Current iteration:4022
Current iteration:4023
Current iteration:4024
Current iteration:4025
Current iteration:4026
Current iteration:4027
Current iteration:4028
Current iteration:4029
Current iteration:4030
Current iteration:4031
Current iteration:4032
Current iteration:4033
Current iteration:4034
Current iteration:4035
Current iteration:4036
Current iteration:4037
Current iteration:4038
Current iteration:4039
Current iteration:4040
Current iteration:4041
Current iteration:4042
Current iteration:4043
Current iteration:4044
Current iteration:4045
Current iteration:4046
Current iteration:4047
Current iteration:4048
Current iteration:4049
Current iteration:4050
Current iteration:4051
Current iteration:4052
Current iteration:4053
Current iteration:4054
Current iteration:4055
Current iteration:4056
Current iteration:4057
Current iteration:4058
Current iteration:4059
Current iteration:4060
Current iteration:4061
Current iteration:4062
Current iteration:4063
Current iteration:4064
Current ite

Current iteration:4383
Current iteration:4384
Current iteration:4385
Current iteration:4386
Current iteration:4387
Current iteration:4388
Current iteration:4389
Current iteration:4390
Current iteration:4391
Current iteration:4392
Current iteration:4393
Current iteration:4394
Current iteration:4395
Current iteration:4396
Current iteration:4397
Current iteration:4398
Current iteration:4399
Current iteration:4400
Current iteration:4401
Current iteration:4402
Current iteration:4403
Current iteration:4404
Current iteration:4405
Current iteration:4406
Current iteration:4407
Current iteration:4408
Current iteration:4409
Current iteration:4410
Current iteration:4411
Current iteration:4412
Current iteration:4413
Current iteration:4414
Current iteration:4415
Current iteration:4416
Current iteration:4417
Current iteration:4418
Current iteration:4419
Current iteration:4420
Current iteration:4421
Current iteration:4422
Current iteration:4423
Current iteration:4424
Current iteration:4425
Current ite

Current iteration:4751
Current iteration:4752
Current iteration:4753
Current iteration:4754
Current iteration:4755
Current iteration:4756
Current iteration:4757
Current iteration:4758
Current iteration:4759
Current iteration:4760
Current iteration:4761
Current iteration:4762
Current iteration:4763
Current iteration:4764
Current iteration:4765
Current iteration:4766
Current iteration:4767
Current iteration:4768
Current iteration:4769
Current iteration:4770
Current iteration:4771
Current iteration:4772
Current iteration:4773
Current iteration:4774
Current iteration:4775
Current iteration:4776
Current iteration:4777
Current iteration:4778
Current iteration:4779
Current iteration:4780
Current iteration:4781
Current iteration:4782
Current iteration:4783
Current iteration:4784
Current iteration:4785
Current iteration:4786
Current iteration:4787
Current iteration:4788
Current iteration:4789
Current iteration:4790
Current iteration:4791
Current iteration:4792
Current iteration:4793
Current ite

Current iteration:5116
Current iteration:5117
Current iteration:5118
Current iteration:5119
Current iteration:5120
Current iteration:5121
Current iteration:5122
Current iteration:5123
Current iteration:5124
Current iteration:5125
Current iteration:5126
Current iteration:5127
Current iteration:5128
Current iteration:5129
Current iteration:5130
Current iteration:5131
Current iteration:5132
Current iteration:5133
Current iteration:5134
Current iteration:5135
Current iteration:5136
Current iteration:5137
Current iteration:5138
Current iteration:5139
Current iteration:5140
Current iteration:5141
Current iteration:5142
Current iteration:5143
Current iteration:5144
Current iteration:5145
Current iteration:5146
Current iteration:5147
Current iteration:5148
Current iteration:5149
Current iteration:5150
Current iteration:5151
Current iteration:5152
Current iteration:5153
Current iteration:5154
Current iteration:5155
Current iteration:5156
Current iteration:5157
Current iteration:5158
Current ite

## The next step is taken so that we can look at data using cause names rather than cause IDs and age group names rather than age group IDs

In [51]:
dalys_haq_sdi_answers = dalys_haq_sdi_answers.merge(dalys_haq_sdi_withCauses_distinct, how='left')

In [52]:
dalys_haq_sdi_answers.head()

cause_id  age_group_id  sex_id     intercept  intercept_pValue  \
0     297.0           2.0     1.0  6.724248e-07      8.726134e-59   
1     297.0           2.0     2.0  1.145769e-06      4.600007e-67   
2     297.0           3.0     1.0  7.689541e-06      9.615998e-89   
3     297.0           3.0     2.0  1.321202e-05      4.268129e-96   
4     297.0           4.0     1.0  2.416194e-02     1.725649e-120   

   haq_over_frontier_coef  haq_over_frontier_coef_pValue  logit_sdi_coef  \
0            8.451967e-10                       0.068245   -4.293437e-07   
1            1.975339e-09                       0.006805   -8.410141e-07   
2            6.192519e-09                       0.131089   -5.069430e-06   
3            9.515223e-09                       0.154983   -8.830539e-06   
4           -1.212206e-05                       0.248310   -1.350129e-02   

   logit_sdi_coef_pValue        r2  age_group_name    cause_name     sex  
0           1.221740e-11  0.146927  Early Neonatal  Tuberculosis    Male  
1           5.367936e-17  0.193144  Early Neonatal  Tuberculosis  Female  
2           3.668217e-19  0.273794   Late Neonatal  Tuberculosis    Male  
3           1.713496e-21  0.308769   Late Neonatal  Tuberculosis  Female  
4           1.593250e-20  0.406797   Post Neonatal  Tuberculosis    Male

## Filtering to keep only columns we're interested in, dropping NaNs, and then keeping only those rows where the coefficient for HAQ_over_Frontier as well SDI is statistically significant according to our model above

In [53]:
dalys_haq_sdi_answers_out = dalys_haq_sdi_answers[['cause_name', 'age_group_name', 'sex', \
                                                   'intercept', 'intercept_pValue', \
                                                   'haq_over_frontier_coef', 'haq_over_frontier_coef_pValue',\
                                                   'logit_sdi_coef', 'logit_sdi_coef_pValue', 'r2']]

In [54]:
dalys_haq_sdi_answers_out.head()

cause_name  age_group_name     sex     intercept  intercept_pValue  \
0  Tuberculosis  Early Neonatal    Male  6.724248e-07      8.726134e-59   
1  Tuberculosis  Early Neonatal  Female  1.145769e-06      4.600007e-67   
2  Tuberculosis   Late Neonatal    Male  7.689541e-06      9.615998e-89   
3  Tuberculosis   Late Neonatal  Female  1.321202e-05      4.268129e-96   
4  Tuberculosis   Post Neonatal    Male  2.416194e-02     1.725649e-120   

   haq_over_frontier_coef  haq_over_frontier_coef_pValue  logit_sdi_coef  \
0            8.451967e-10                       0.068245   -4.293437e-07   
1            1.975339e-09                       0.006805   -8.410141e-07   
2            6.192519e-09                       0.131089   -5.069430e-06   
3            9.515223e-09                       0.154983   -8.830539e-06   
4           -1.212206e-05                       0.248310   -1.350129e-02   

   logit_sdi_coef_pValue        r2  
0           1.221740e-11  0.146927  
1           5.367936e-17  0.193144  
2           3.668217e-19  0.273794  
3           1.713496e-21  0.308769  
4           1.593250e-20  0.406797

In [55]:
dalys_haq_sdi_answers_out.shape

(5427, 10)

In [56]:
dalys_haq_sdi_answers_out = dalys_haq_sdi_answers_out.dropna()

In [57]:
dalys_haq_sdi_answers_out.shape

(5426, 10)

In [58]:
dalys_haq_sdi_answers_out_significant = dalys_haq_sdi_answers[dalys_haq_sdi_answers['haq_over_frontier_coef_pValue'] < 0.05]
dalys_haq_sdi_answers_out_significant = dalys_haq_sdi_answers_out_significant[dalys_haq_sdi_answers_out_significant['logit_sdi_coef_pValue'] < 0.05]
             

In [59]:
dalys_haq_sdi_answers_out_significant.shape

(3178, 13)

In [60]:
dalys_haq_sdi_answers_out_significant['cause_id'] = dalys_haq_sdi_answers_out_significant['cause_id'].astype('int64')

In [61]:
dalys_haq_sdi_answers_out_significant['age_group_id'] = dalys_haq_sdi_answers_out_significant['age_group_id'].astype('int64')

In [62]:
dalys_haq_sdi_answers_out_significant['sex_id'] = dalys_haq_sdi_answers_out_significant['sex_id'].astype('int64')

In [63]:
dalys_haq_sdi_answers_out_significant.head()

cause_id  age_group_id  sex_id  intercept  intercept_pValue  \
1       297             2       2   0.000001      4.600007e-67   
6       297             5       1   0.004311      1.423142e-84   
7       297             5       2   0.004335      1.927075e-82   
8       297             6       1   0.001072      1.570181e-92   
9       297             6       2   0.001184     9.220505e-132   

   haq_over_frontier_coef  haq_over_frontier_coef_pValue  logit_sdi_coef  \
1            1.975339e-09                       0.006805   -8.410141e-07   
6            6.905744e-06                       0.003655   -3.822178e-03   
7            9.716194e-06                       0.000066   -4.265410e-03   
8           -1.238627e-06                       0.026167   -4.816614e-04   
9           -1.064624e-06                       0.027662   -5.747071e-04   

   logit_sdi_coef_pValue        r2  age_group_name    cause_name     sex  
1           5.367936e-17  0.193144  Early Neonatal  Tuberculosis  Female  
6           1.671519e-30  0.353012          1 to 4  Tuberculosis    Male  
7           1.532920e-35  0.362330          1 to 4  Tuberculosis  Female  
8           2.271153e-10  0.305315          5 to 9  Tuberculosis    Male  
9           6.350739e-18  0.418625          5 to 9  Tuberculosis  Female

## Doing some analysis to answer some of the questions related to the data produced by the regressions

In [64]:
len(dalys_haq_sdi_answers_out_significant[(dalys_haq_sdi_answers_out_significant['haq_over_frontier_coef'] > 0)\
                                          | (dalys_haq_sdi_answers_out_significant['logit_sdi_coef'] > 0)])

2460

In [65]:
len(dalys_haq_sdi_answers_out_significant[dalys_haq_sdi_answers_out_significant['haq_over_frontier_coef'] > 0])

1158

In [66]:
1158/2460

0.47073170731707314

In [67]:
import numpy as np
np.unique(dalys_haq_sdi_answers_out_significant['age_group_id'])

array([  2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,  13,  14,
        15,  16,  17,  18,  19,  20,  30,  31,  32, 235])

In [68]:
np.unique(dalys_haq_sdi_answers_out_significant['age_group_name'])

array(['1 to 4', '10 to 14', '15 to 19', '20 to 24', '25 to 29',
       '30 to 34', '35 to 39', '40 to 44', '45 to 49', '5 to 9',
       '50 to 54', '55 to 59', '60 to 64', '65 to 69', '70 to 74',
       '75 to 79', '80 to 84', '85 to 89', '90 to 94', '95 plus',
       'Early Neonatal', 'Late Neonatal', 'Post Neonatal'], dtype=object)

In [69]:
dalys_haq_sdi_answers_out_significant_withoutOld = \
dalys_haq_sdi_answers_out_significant[dalys_haq_sdi_answers_out_significant['age_group_id'] < 20.0 ]

In [70]:
len(dalys_haq_sdi_answers_out_significant_withoutOld)

2453

In [71]:
len(dalys_haq_sdi_answers_out_significant_withoutOld[(dalys_haq_sdi_answers_out_significant_withoutOld['haq_over_frontier_coef'] > 0)\
                                          | (dalys_haq_sdi_answers_out_significant_withoutOld['logit_sdi_coef'] > 0)])

1882

In [72]:
len(dalys_haq_sdi_answers_out_significant_withoutOld[dalys_haq_sdi_answers_out_significant_withoutOld['haq_over_frontier_coef'] > 0])

816

In [73]:
816/1882

0.43358129649309246

In [98]:
dalys_haq_sdi_answers_out_significant_withoutOld.to_csv('/ihme/homes/arjuns13/notebooks/Documents/Data/dalys_haq_sdi_allcauses_ageStd_without74plus_14Jun19.csv')

## Segregating for men and women

In [74]:
dalys_haq_sdi_answers_out_significant_wihout74plus_male = \
dalys_haq_sdi_answers_out_significant_withoutOld[dalys_haq_sdi_answers_out_significant_withoutOld['sex_id']==1]

In [75]:
dalys_haq_sdi_answers_out_significant_wihout74plus_male.head()

cause_id  age_group_id  sex_id  intercept  intercept_pValue  \
6        297             5       1   0.004311      1.423142e-84   
8        297             6       1   0.001072      1.570181e-92   
10       297             7       1   0.001143     1.138196e-123   
14       297             9       1   0.008794     3.618034e-128   
30       297            17       1   0.041626     3.612539e-157   

    haq_over_frontier_coef  haq_over_frontier_coef_pValue  logit_sdi_coef  \
6                 0.000007                       0.003655       -0.003822   
8                -0.000001                       0.026167       -0.000482   
10               -0.000001                       0.014407       -0.000534   
14               -0.000008                       0.026408       -0.004254   
30               -0.000046                       0.002110       -0.018986   

    logit_sdi_coef_pValue        r2 age_group_name    cause_name   sex  
6            1.671519e-30  0.353012         1 to 4  Tuberculosis  Male  
8            2.271153e-10  0.305315         5 to 9  Tuberculosis  Male  
10           1.758747e-15  0.396821       10 to 14  Tuberculosis  Male  
14           3.278215e-17  0.410270       20 to 24  Tuberculosis  Male  
30           3.678776e-20  0.479931       60 to 64  Tuberculosis  Male

In [76]:
dalys_haq_sdi_answers_out_significant_wihout74plus_female = \
dalys_haq_sdi_answers_out_significant_withoutOld[dalys_haq_sdi_answers_out_significant_withoutOld['sex_id']==2]

In [77]:
dalys_haq_sdi_answers_out_significant_wihout74plus_male.shape

(1192, 13)

In [78]:
dalys_haq_sdi_answers_out_significant_wihout74plus_female.head()

cause_id  age_group_id  sex_id  intercept  intercept_pValue  \
1        297             2       2   0.000001      4.600007e-67   
7        297             5       2   0.004335      1.927075e-82   
9        297             6       2   0.001184     9.220505e-132   
11       297             7       2   0.001232     1.197681e-151   
13       297             8       2   0.003404     9.868881e-128   

    haq_over_frontier_coef  haq_over_frontier_coef_pValue  logit_sdi_coef  \
1             1.975339e-09                       0.006805   -8.410141e-07   
7             9.716194e-06                       0.000066   -4.265410e-03   
9            -1.064624e-06                       0.027662   -5.747071e-04   
11           -9.197531e-07                       0.042690   -6.251903e-04   
13           -5.666051e-06                       0.000071   -1.269826e-03   

    logit_sdi_coef_pValue        r2  age_group_name    cause_name     sex  
1            5.367936e-17  0.193144  Early Neonatal  Tuberculosis  Female  
7            1.532920e-35  0.362330          1 to 4  Tuberculosis  Female  
9            6.350739e-18  0.418625          5 to 9  Tuberculosis  Female  
11           3.387027e-23  0.471737        10 to 14  Tuberculosis  Female  
13           5.897368e-11  0.396572        15 to 19  Tuberculosis  Female

In [79]:
dalys_haq_sdi_answers_out_significant_wihout74plus_female.shape

(1261, 13)

In [80]:
final_table_male = dalys_haq_sdi_answers_out_significant_wihout74plus_male[['cause_id', 'cause_name',\
                                                                            'age_group_id', 'age_group_name',\
                                                                            'haq_over_frontier_coef']]                                              

In [81]:
final_table_male.head()

cause_id    cause_name  age_group_id age_group_name  \
6        297  Tuberculosis             5         1 to 4   
8        297  Tuberculosis             6         5 to 9   
10       297  Tuberculosis             7       10 to 14   
14       297  Tuberculosis             9       20 to 24   
30       297  Tuberculosis            17       60 to 64   

    haq_over_frontier_coef  
6                 0.000007  
8                -0.000001  
10               -0.000001  
14               -0.000008  
30               -0.000046

In [82]:
final_table_male = pd.pivot_table(final_table_male,\
                                  values='haq_over_frontier_coef', index=['cause_id', 'cause_name'],\
                                  columns=['age_group_id', 'age_group_name'], fill_value='NA')

In [83]:
final_table_male

age_group_id                                                            2   \
age_group_name                                              Early Neonatal   
cause_id cause_name                                                          
297      Tuberculosis                                                   NA   
298      HIV/AIDS                                                       NA   
302      Diarrheal diseases                                     0.00190387   
321      Other intestinal infectious diseases                           NA   
322      Lower respiratory infections                                   NA   
328      Upper respiratory infections                                   NA   
329      Otitis media                                          3.98339e-06   
332      Meningitis                                            0.000817558   
337      Encephalitis                                         -6.10851e-05   
338      Diphtheria                                                     NA   
339      Whooping cough                                                 NA   
340      Tetanus                                                 0.0011485   
341      Measles                                                        NA   
342      Varicella and herpes zoster                                    NA   
344      Neglected tropical diseases and malaria               2.74639e-05   
381      Neonatal preterm birth                                 -0.0369472   
382      Neonatal encephalopathy due to birth asphyxia a...     -0.0148425   
383      Neonatal sepsis and other neonatal infections         -0.00943292   
384      Hemolytic disease and other neonatal jaundice          -0.0027362   
385      Other neonatal disorders                               -0.0265131   
387      Protein-energy malnutrition                          -1.04053e-05   
388      Iodine deficiency                                              NA   
389      Vitamin A deficiency                                           NA   
390      Dietary iron deficiency                              -5.94795e-06   
391      Other nutritional deficiencies                                 NA   
393      Sexually transmitted infections excluding HIV         -0.00506882   
400      Acute hepatitis                                                NA   
405      Leprosy                                                        NA   
408      Other unspecified infectious diseases                          NA   
411      Esophageal cancer                                              NA   
...                                                                    ...   
588      Acute glomerulonephritis                                       NA   
589      Chronic kidney disease                                         NA   
594      Urinary diseases and male infertility                 8.20434e-05   
613      Hemoglobinopathies and hemolytic anemias                       NA   
619      Endocrine, metabolic, blood, and immune disorders    -0.000146727   
627      Rheumatoid arthritis                                           NA   
628      Osteoarthritis                                                 NA   
630      Low back pain                                                  NA   
631      Neck pain                                                      NA   
632      Gout                                                           NA   
639      Other musculoskeletal disorders                                NA   
641      Congenital birth defects                                       NA   
653      Skin and subcutaneous diseases                                 NA   
669      Sense organ diseases                                 -1.13147e-06   
680      Oral disorders                                                 NA   
689      Road injuries                                                  NA   
695      Other transport injuries                                       NA   
697      Falls        

In [84]:
final_table_male.shape

(129, 18)

In [85]:
type(final_table_male)

pandas.core.frame.DataFrame

In [86]:
final_table_female = dalys_haq_sdi_answers_out_significant_wihout74plus_female[['cause_id', 'cause_name',\
                                                                            'age_group_id', 'age_group_name',\
                                                                            'haq_over_frontier_coef']]                                              

In [87]:
final_table_female.head()

cause_id    cause_name  age_group_id  age_group_name  \
1        297  Tuberculosis             2  Early Neonatal   
7        297  Tuberculosis             5          1 to 4   
9        297  Tuberculosis             6          5 to 9   
11       297  Tuberculosis             7        10 to 14   
13       297  Tuberculosis             8        15 to 19   

    haq_over_frontier_coef  
1             1.975339e-09  
7             9.716194e-06  
9            -1.064624e-06  
11           -9.197531e-07  
13           -5.666051e-06

In [88]:
final_table_female.shape

(1261, 5)

In [89]:
final_table_female = pd.pivot_table(final_table_female,\
                                    values='haq_over_frontier_coef', index=['cause_id', 'cause_name'],\
                                    columns=['age_group_id', 'age_group_name'], fill_value='NA')

In [90]:
final_table_female

age_group_id                                                            2   \
age_group_name                                              Early Neonatal   
cause_id cause_name                                                          
297      Tuberculosis                                          1.97534e-09   
298      HIV/AIDS                                                       NA   
302      Diarrheal diseases                                             NA   
321      Other intestinal infectious diseases                           NA   
322      Lower respiratory infections                                   NA   
328      Upper respiratory infections                                   NA   
329      Otitis media                                          6.32508e-06   
332      Meningitis                                            0.000804025   
337      Encephalitis                                         -3.53239e-05   
338      Diphtheria                                                     NA   
339      Whooping cough                                                 NA   
340      Tetanus                                                 0.0024369   
341      Measles                                                        NA   
342      Varicella and herpes zoster                                    NA   
344      Neglected tropical diseases and malaria                        NA   
367      Maternal hemorrhage                                            NA   
368      Maternal sepsis and other maternal infections                  NA   
369      Maternal hypertensive disorders                                NA   
370      Maternal obstructed labor and uterine rupture                  NA   
374      Ectopic pregnancy                                              NA   
375      Indirect maternal deaths                                       NA   
379      Other maternal disorders                                       NA   
381      Neonatal preterm birth                                 -0.0325165   
382      Neonatal encephalopathy due to birth asphyxia a...    -0.00949222   
383      Neonatal sepsis and other neonatal infections         -0.00847816   
384      Hemolytic disease and other neonatal jaundice                  NA   
385      Other neonatal disorders                               -0.0194897   
387      Protein-energy malnutrition                                    NA   
388      Iodine deficiency                                              NA   
389      Vitamin A deficiency                                           NA   
...                                                                    ...   
594      Urinary diseases and male infertility                 2.56006e-05   
603      Gynecological diseases                                         NA   
613      Hemoglobinopathies and hemolytic anemias                       NA   
619      Endocrine, metabolic, blood, and immune disorders    -0.000155548   
627      Rheumatoid arthritis                                           NA   
628      Osteoarthritis                                                 NA   
630      Low back pain                                                  NA   
631      Neck pain                                                      NA   
632      Gout                                                           NA   
639      Other musculoskeletal disorders                                NA   
641      Congenital birth defects                                       NA   
653      Skin and subcutaneous diseases                                 NA   
669      Sense organ diseases                                  -5.1302e-07   
680      Oral disorders                                                 NA   
689      Road injuries                                                  NA   
695      Other transport injuries                             -1.41265e-05   
697      Falls                                                          NA   
698      Drowning     

In [91]:
final_table_female.shape

(136, 18)

In [92]:
type(final_table_female)

pandas.core.frame.DataFrame

## Changed the below to Markdown type rather than code to make sure, nothing gets over-written by mistake

In [93]:
final_table_female.to_csv('/ihme/homes/arjuns13/notebooks/Documents/Data/dalys_haq_sdi_females_final_table_ageStd_14Jun19.csv')

In [94]:
final_table_female.to_csv('J:\\Project\\Cost_Effectiveness\\NPC\\Regression_Analysis\\dalys_haq_sdi_females_final_table_ageStd_14Jun19.csv')

In [95]:
final_table_male.to_csv('/ihme/homes/arjuns13/notebooks/Documents/Data/dalys_haq_sdi_males_final_table_ageStd_14Jun19.csv')

In [96]:
final_table_male.to_csv('J:\\Project\\Cost_Effectiveness\\NPC\\Regression_Analysis\\dalys_haq_sdi_males_final_table_ageStd_14Jun19.csv')